lib/chef/event_dispatch/dispatcher.rb

```diff
@@ -28,6 +28,9 @@ def register(subscriber)
       # Define a method that will be forwarded to all
       def self.def_forwarding_method(method_name)
         define_method(method_name) do |*args|
+          if method_name == :deprecation && args.size == 1
```

Comment:
If we do really need a special case, let's not do it inside the metaprogramming. We try to keep the logic inside the metaprogramming stuff as simple as possible. We could instead just define the method normally after the mass-defining all the normally-behaved methods.